config/config.go

```diff
@@ -354,6 +369,36 @@ func setupTransport() {
 	}
 }
 
+func setupCSRFHandler() {
```

Comment:
Can you refactor this into a separate file in the `config` directory?  I fear `config.go` is already too large, adding a bunch of web-related imports only makes it more complicated to read/review.